File: Leetcode/JS/Leetcode1154-一年中的第几天.js
var dayOfYear = function(date) {
    let nums = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    const y = +date.slice(0,4);
    const m = +date.slice(5,7);
    const d = +date.slice(8)
    let is_leap = false;
    if(y % 400 == 0 || (y % 4 ==0 && y % 100 != 0)) is_leap = true;
    let ans = 0;
    for(let i = 0;i < m - 1;i++){
        ans += nums[i];
        if(is_leap && i == 1) {
            ans += 1;
        }
    }
    return ans + d;
};

File: Leetcode/PY/Leetcode1154-一年中的第几天.py
class Solution:
    def dayOfYear(self, date: str) -> int:
        arr = date.split('-')
        y,m,d = int(arr[0]),int(arr[1]),int(arr[2])
        nums = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
        is_leap = True if y % 400 == 0 or (y % 4 == 0 and y % 100 != 0) else False
        ans = 0
        for i in range(m - 1):
            ans += nums[i]
            if is_leap and i == 1:
                ans += 1
        return ans + d
